Mathlib/Init/Order/Defs.lean

```diff
@@ -1,411 +0,0 @@
-/-
-Copyright (c) 2016 Microsoft Corporation. All rights reserved.
-Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Leonardo de Moura
--/
-import Mathlib.Init.Algebra.Classes
-import Mathlib.Data.Ordering.Basic
-import Mathlib.Tactic.SplitIfs
-import Mathlib.Tactic.TypeStar
-import Batteries.Classes.Order
-
-/-!
-# Note about `Mathlib/Init/`
-The files in `Mathlib/Init` are leftovers from the port from Mathlib3.
-(They contain content moved from lean3 itself that Mathlib needed but was not moved to lean4.)
-
-We intend to move all the content of these files out into the main `Mathlib` directory structure.
-Contributions assisting with this are appreciated.
-
-# Orders
-
-Defines classes for preorders, partial orders, and linear orders
-and proves some basic lemmas about them.
--/
-
-universe u
-variable {α : Type u}
-
-section Preorder
-
-/-!
-### Definition of `Preorder` and lemmas about types with a `Preorder`
--/
-
-/-- A preorder is a reflexive, transitive relation `≤` with `a < b` defined in the obvious way. -/
-class Preorder (α : Type u) extends LE α, LT α where
-  le_refl : ∀ a : α, a ≤ a
-  le_trans : ∀ a b c : α, a ≤ b → b ≤ c → a ≤ c
-  lt := fun a b => a ≤ b ∧ ¬b ≤ a
-  lt_iff_le_not_le : ∀ a b : α, a < b ↔ a ≤ b ∧ ¬b ≤ a := by intros; rfl
-
-variable [Preorder α]
-
-/-- The relation `≤` on a preorder is reflexive. -/
-@[refl]
-theorem le_refl : ∀ a : α, a ≤ a :=
-  Preorder.le_refl
-
-/-- A version of `le_refl` where the argument is implicit -/
-theorem le_rfl {a : α} : a ≤ a :=
-  le_refl a
-
-/-- The relation `≤` on a preorder is transitive. -/
-@[trans]
-theorem le_trans : ∀ {a b c : α}, a ≤ b → b ≤ c → a ≤ c :=
-  Preorder.le_trans _ _ _
-
-theorem lt_iff_le_not_le : ∀ {a b : α}, a < b ↔ a ≤ b ∧ ¬b ≤ a :=
-  Preorder.lt_iff_le_not_le _ _
-
-theorem lt_of_le_not_le : ∀ {a b : α}, a ≤ b → ¬b ≤ a → a < b
-  | _a, _b, hab, hba => lt_iff_le_not_le.mpr ⟨hab, hba⟩
-
-theorem le_not_le_of_lt : ∀ {a b : α}, a < b → a ≤ b ∧ ¬b ≤ a
-  | _a, _b, hab => lt_iff_le_not_le.mp hab
```

Comment:
If you want to remove random theorems, please do that in a separate PR, after they've been deprecated for a while

Reply:
This one is a useless combination of other lemmas and uses are anecdotic. Am happy to see it restored deprecated, though